conda_forge_tick/update_recipe/version.py

```diff
@@ -625,9 +676,103 @@ def _update_version_feedstock_dir_containerized(feedstock_dir, version, hash_typ
     return data["updated"], data["errors"]
 
 
-def update_version(raw_meta_yaml, version, hash_type="sha256"):
+def update_version_v1(
+    feedstock_dir: str, version: str, hash_type: str
+) -> (str | None, set[str]):
     """Update the version in a recipe.
 
+    Parameters
+    ----------
+    feedstock_dir : str
+        The feedstock directory to update.
+    version : str
+        The new version of the recipe.
+    hash_type : str
+        The kind of hash used on the source.
+
+    Returns
+    -------
+    recipe_text : str or None
+        The text of the updated recipe.yaml. Will be None if there is an error.
+    errors : set of str
+    """
+    # extract all the URL sources from a given recipe / feedstock directory
+    from rattler_build_conda_compat.loader import load_yaml
+    from rattler_build_conda_compat.recipe_sources import render_all_sources
+
+    feedstock_dir = Path(feedstock_dir)
+    recipe_path = feedstock_dir / "recipe" / "recipe.yaml"
+    recipe_yaml = load_yaml(recipe_path.read_text())
+    variants = feedstock_dir.glob(".ci_support/*.yaml")
+    # load all variants
+    variants = [load_yaml(variant.read_text()) for variant in variants]
+    if not len(variants):
+        # if there are no variants, then we need to add an empty one
+        variants = [{}]
+
+    rendered_sources = render_all_sources(
+        recipe_yaml, variants, override_version=version
+    )
+
+    recipe_text = recipe_path.read_text()
```

Comment:
Please move that line up a bit to where `recipe_yaml` is calculated from it.

Reply:
feel free to just push to this PR :)

Reply:
Yeah I'm happy to wait on this until all comments are addressed. As usual, the folks wanting things done have the onus to do them. I am not 100% happy with how the interactions and pushing in both directions (so me inlcuded) have happened with this PR and v0 -> v1 tasks in general. We need to find a better way to communicate and balance things.

Reply:
> feel free to just push to this PR :)

Alright, didn't want to assume, but fine by me. 👍 

> I am not 100% happy with how the interactions and pushing in both directions (so me inlcuded) have happened with this PR

Do you want to elaborate on that? For me it's been pretty much standard review - the work in itself is highly appreciated, and yet at the same time, we can ensure through review-response iterations that newly added code is consistent and otherwise of a high quality before merging.

Reply:
Sure! There has been some back and forth around skipping edge cases / excluding things just to get PRs done. I've also felt like i've been put in the position of blocking half-completed PRs. I'd rather not have to push back at each step and instead have things fully implemented up front. 

I also think I have not been the most responsive on PRs due to my frustration and wanting to avoid the interactions.